oat-sa/tao-core
# Feature/ADF-1752/Confirm category propagation

### Related to: https://oat-sa.atlassian.net/browse/ADF-1752

### Summary

Set the correct style to the tags in a selec2 selector when a partial representation is declared.

### Details

When a tag applies to a part only of the contained items, it must have a style that reflects this state.

With the legacy theme, this is the case. But with the Solar UI, it was not applied.

- Legacy
![image](https://github.com/user-attachments/assets/6126b8fd-e828-4b6e-a0a5-5a508a677ddd)

- Solar UI
    - before:
    ![image](https://github.com/user-attachments/assets/d7b2b163-a9fe-4655-b724-e7549a723126)
    - after
    ![image](https://github.com/user-attachments/assets/16354aae-d042-4990-9b13-f57e7f222eee)


### How to test

- checkout the branch: `git checkout -t origin/feature/ADF-1752/confirm-category-propagation`
- open TAO backoffice, then author a test with multiple items in it
- set a custom category to a test part or a section
- from an item, remove the category
- go back to the test part or section and see the style
- check with and without the Solar UI style (can be made by setting and removing the class `solar-design` on the body tag)
